stl/inc/xspan

```diff
@@ -0,0 +1,238 @@
+#if _CONTAINER_DEBUG_LEVEL != 0
+            _STL_VERIFY(_Size == _Extent, "Mismatch between dynamic extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        constexpr index_type size() const noexcept {
+            return _Extent;
+        }
+    };
+
+    template <>
+    class _Extent_type<dynamic_extent> {
+    public:
+        using index_type = size_t;
+
+        constexpr _Extent_type() noexcept : _Mysize(index_type(0)) {}
+
+        template <index_type _OtherExtent>
+        explicit constexpr _Extent_type(_Extent_type<_OtherExtent> _Ext) noexcept : _Mysize(_Ext.size()) {}
+
+        explicit constexpr _Extent_type(index_type _Size) noexcept : _Mysize(_Size) {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Mysize >= 0, "Mismatch between dynamic extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        constexpr index_type size() const noexcept {
+            return _Mysize;
+        }
+
+    private:
+        index_type _Mysize;
+    };
+} // namespace details
+
+// [views.span], class template span
+template <class _Ty, size_t _Extent = dynamic_extent>
+class span {
+public:
+    using element_type    = _Ty;
+    using value_type      = remove_cv_t<_Ty>;
+    using index_type      = size_t;
+    using difference_type = ptrdiff_t;
+    using pointer         = _Ty*;
+    using const_pointer   = const _Ty*;
+    using reference       = _Ty&;
+    using const_reference = const _Ty&;
+    // using iterator               = _Span_iterator<pointer>;
+    // using const_iterator         = _Span_const_iterator<const_pointer>;
+    // using reverse_iterator       = _STD reverse_iterator<iterator>;
+    // using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+    // [span.cons], span constructors, copy, assignment, and destructor
+    constexpr span() noexcept : _Mypair(_Zero_then_variadic_args_t(), nullptr) {}
+
+    constexpr span(pointer _Ptr, index_type _Count) noexcept : _Mypair(_One_then_variadic_args_t(), _Count, _Ptr) {}
+
+    constexpr span(pointer _First, pointer _Last) noexcept
+        : _Mypair(_One_then_variadic_args_t(), static_cast<index_type>(distance(_First, _Last)), _First) {}
+
+    template <size_t _Size>
+    constexpr span(element_type (&_Data)[_Size]) noexcept
+        : _Mypair(_One_then_variadic_args_t(), _Size, addressof(_Data[0])) {}
+
+    template <size_t _Size>
+    constexpr span(array<value_type, _Size>& _Data) noexcept
+        : _Mypair(_One_then_variadic_args_t(), _Size, _Data.data()) {}
+
+    template <size_t _Size>
+    constexpr span(const array<value_type, _Size>& _Data) noexcept
+        : _Mypair(_One_then_variadic_args_t(), _Size, _Data.data()) {}
+
+    template <class _OtherTy, size_t _OtherExtent>
+    constexpr span(const span<_OtherTy, _OtherExtent>& _Other,
+        enable_if_t<is_convertible_v<_OtherTy (*)[], element_type (*)[]>, nullptr_t> = nullptr) noexcept

```

Comment:
We use `enable_if_t` in template parameters, never function parameters. Our modern form is `enable_if_t<CONDITION, int> = 0`. Occurs below.

Reply:
So this would be changed to 

```cpp
    template <class _OtherTy, size_t _OtherExtent, 
        enable_if_t<is_convertible_v<_OtherTy (*)[], element_type (*)[]>, int> = 0>
    constexpr span(const span<_OtherTy, _OtherExtent>& _Other) noexcept
        : _Mydata(_OtherExtent, _Other.data()) {}
```